xarray/core/variable.py

```diff
@@ -893,16 +893,27 @@ def reduce(self, func, dim=None, axis=None, keep_attrs=False,
         if dim is not None and axis is not None:
             raise ValueError("cannot supply both 'axis' and 'dim' arguments")
 
+        if 'cum' in func.__name__:
```

Comment:
@shoyer, this obviously changes the comments below because we need this to work for both dataset and dataarray and I need to verify the existing implementation does just that.